plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/MachineProviderImpl.java

```diff
@@ -573,13 +575,16 @@ private void addSystemWideContainerSettings(String workspaceId,
             volumes = devMachineSystemVolumes;
 
             env = new HashMap<>(devMachineEnvVariables);
-            env.put(DockerInstanceRuntimeInfo.CHE_WORKSPACE_ID, workspaceId);
             env.put(DockerInstanceRuntimeInfo.USER_TOKEN, getUserToken(workspaceId));
         } else {
             portsToExpose = commonMachinePortsToExpose;
-            env = commonMachineEnvVariables;
+            env = new HashMap<>(commonMachineEnvVariables);
             volumes = commonMachineSystemVolumes;
         }
+        // register workspace ID and Machine Name
+        env.put(DockerInstanceRuntimeInfo.CHE_WORKSPACE_ID, workspaceId);
```

Comment:
Looks like it is not covered by tests. Some tests check WORKSPACE_ID variable setting. Behavior has changed so, please, change tests.